spec/controllers/oauth/applications_controller_spec.rb

```diff
@@ -160,13 +162,15 @@ module Oauth
     it "should not let a user get the list of his applications" do
       controller.sign_in! user
       get(:index)
-      expect(response).to have_http_status :forbidden
+      expect(response).to have_http_status :success
+      expect(response.body).to eq('')
```

Comment:
Why doesn't the trusted application show up (since `trusted_application_admin.owner.add_member(user)` is run)

Reply:
This was caused by the improper implementation of admin_authentication. It is now returning a 403 as expected if the user is neither an admin nor an oauth admin.